Molar fractions and state passed to SolidSolution

bobmyhill opened the pull request:
The optional argument *molar_fraction* is added to the __init__ of the SolidSolution class. This allows the user to set the [composition] of the solid solution with a single line rather than two. The order of variables is chosen for compatibility with older SolidSolution classes.

bobmyhill commented:
What do you think? Especially @tjhei as our arbiter of syntax :)
I split the commits relevant to molar fraction and those relevant to state, just in case you didn't like the latter. I think this will make for much more compact code. I was considering doing the same for the mineral class (including [P,T] as an __init__ argument), but decided to wait to see what you thought of this first.

ian-r-rose commented:
I don't have super strong feelings about it.
Overall, I would say that putting *molar fraction* in the initializer makes good sense, but I am less enamored with putting the P,T state in the initializer, as this is something that is likely to change all the time, and feels less appropriate to it.

bobmyhill commented:
Yup, I'm in two minds about the state part as well. I'm happy to remove this part ... it didn't take long to write.

tjhei commented:
I agree with Ian, what is the reason to include the state?

bobmyhill commented:
Ok, so as I was ambivalent about adding state (I added it for code brevity only), and you're both against including it, I've removed those commits from the history. This should be ready to roll.

ian-r-rose commented on burnman/solidsolution.py:
```diff
@@ -27,7 +27,7 @@ class SolidSolution(Mineral):
     """
 
-    def __init__(self, endmembers, solution_model=SolutionModel()):
+    def __init__(self, endmembers, solution_model=SolutionModel(), molar_fraction=None):
```
This seems like a bug.  In some places you are calling it molar_fraction, and in others you are calling it molar_fractions

bobmyhill replied:
Not a bug (I thing I've been consistent in each file). It's just a horrible legacy from our mega commit in December. We deleted the commit where I fixed this. I can change them all to molar_fractions...

ian-r-rose replied:
Ah, I see now.  My mistake.

bobmyhill replied:
Fixed anyway ... I meant to (re)do this a while ago :)

bobmyhill commented:
Can this be merged now?

tjhei commented:
Sorry for the delay. Looks good, tests pass.
